assets/src/edit-story/app/api/apiProvider.js

```diff
@@ -386,6 +387,34 @@ function APIProvider({ children }) {
     [cdnURL, assetsURL]
   );
 
+  // @todo Add paging.
+  const getCustomPageTemplates = useCallback(() => {
+    let apiPath = customPageTemplates;
+    const perPage = 100;
+    apiPath = addQueryArgs(apiPath, {
+      context: 'edit',
+      per_page: perPage,
+      page: 1,
```

Comment:
Why limit to 100? Why not just pass `per_page=-1` here? The [fetch-all-middleware](https://github.com/WordPress/gutenberg/blob/trunk/packages/api-fetch/src/middlewares/fetch-all-middleware.js) will do all the hard work of getting all the page templates for you. The thing with page templates, at most a user would have say around 200/300. Which is only three requests that come back pretty quickly. 

-1 Should be avoid for something like users, as sites can have 40K, it is 400 requests. But in this case, I think -1 should be fine. Thoughts?

Reply:
Hm, good idea. I'll have to test this more, probably will do it within #7271 though, `100` is just temporary.